calico_node/tests/st/utils/workload.py

```diff
@@ -51,20 +54,72 @@ def __init__(self, host, name, image="busybox", network="bridge", ip=None, label
         """
         self.host = host
         self.name = name
+        self.network = network
 
         lbl_args = ""
         for label in labels:
             lbl_args += " --label %s" % (label)
 
         ip_option = ("--ip %s" % ip) if ip else ""
-        command = "docker run -tid --name %s --net %s %s %s %s" % \
-                  (name, network, lbl_args, ip_option, image)
+        command = "docker run -tid --name %s --net=none %s %s" % \
+                  (name, lbl_args, image)
+        # Change to --net none here, and use Calico CNI instead.
         docker_run_wl = partial(host.execute, command)
         retry_until_success(docker_run_wl)
-        self.ip = host.execute(
-            "docker inspect --format "
-            "'{{.NetworkSettings.Networks.%s.IPAddress}}' %s"
-            % (network, name))
+        #self.ip = host.execute(
+        #    "docker inspect --format "
+        #    "'{{.NetworkSettings.Networks.%s.IPAddress}}' %s"
+        #    % (network, name))
+        self.run_cni("ADD", ip=ip)
+
+    def run_cni(self, add_or_del, ip=None):
+        adding = (add_or_del == "ADD")
+        workload_pid = self.host.execute("docker inspect --format '{{.State.Pid}}' %s" % self.name)
+        container_id = self.host.execute("docker inspect --format '{{.Id}}' %s" % self.name)
+        ip_json = (",\"args\":{\"ip\":\"%s\"}" % ip) if (ip and adding) else ""
```

Comment:
I don't _think_ this json will be respected - the calico-ipam CNI plugin will respect the `IP=` env var but doesn't support an args field AFAICT: https://github.com/projectcalico/cni-plugin/blob/master/utils/types.go#L49

Reply:
Agreed. But IIUC the CNI_ARGS usage is already deprecated, and we ought soon to be changing our plugin to honor the JSON version instead. So it seems sensible to me for our test code to specify both formats.

Reply:
Sounds OK. I would have thought we should just switch this line to use the args json once the plugin supported it, but given it's test code I'm fine with leaving both and just removing the IP= env var at a later date.